app/bundles/EmailBundle/Assets/js/email.js

```diff
@@ -330,12 +333,20 @@ Mautic.createNewDynamicContentItem = function(jQueryVariant) {
     var textarea      = itemContainer.find('.editor');
     var firstInput    = itemContainer.find('input[type="text"]').first();
 
-    textarea.froalaEditor(mQuery.extend({}, Mautic.basicFroalaOptions, {
-        // Set custom buttons with separator between them.
-        toolbarSticky: false,
-        toolbarButtons: ['undo', 'redo', '|', 'bold', 'italic', 'underline', 'paragraphFormat', 'fontFamily', 'fontSize', 'color', 'align', 'formatOL', 'formatUL', 'quote', 'clearFormatting', 'token', 'insertLink', 'insertImage', 'insertTable', 'html', 'fullscreen'],
-        heightMin: 100
-    }));
+    if (textarea.hasClass('legacy-builder')) {
+        textarea.froalaEditor(mQuery.extend({}, Mautic.basicFroalaOptions, {
+            // Set custom buttons with separator between them.
+            toolbarSticky: false,
+            toolbarButtons: ['undo', 'redo', '|', 'bold', 'italic', 'underline', 'paragraphFormat', 'fontFamily', 'fontSize', 'color', 'align', 'formatOL', 'formatUL', 'quote', 'clearFormatting', 'token', 'insertLink', 'insertImage', 'insertTable', 'html', 'fullscreen'],
+            heightMin: 100
+        }));
+    }
+
+    if (Mautic.internalDynamicContentItemCreateListeners) {
+        Mautic.internalDynamicContentItemCreateListeners.forEach(function(callback) {
+            callback(textarea);
```

Comment:
I'd like to move Dynamic Content logic to its own JS file after we've merged this PR, to allow for better compatibility with external builders in the future, as well as an consistent API for those. Curious to hear your thoughts on that @adiux